app/js/FileHandler.js

```diff
@@ -73,6 +75,24 @@ async function getFile(bucket, key, opts) {
   }
 }
 
+async function getRedirectUrl(bucket, key, opts) {
+  // if we're doing anything unusual with options, or the request isn't for
+  // one of the default buckets, return null so that we proxy the file
+  opts = opts || {}
+  if (
+    !opts.start &&
+    !opts.end &&
+    !opts.format &&
+    !opts.style &&
+    Object.values(Settings.filestore.stores).includes(bucket) &&
+    Settings.filestore.allowRedirects
```

Comment:
Maybe check for opts being completely empty (Object.keys(opts).length === 0)? I think we can only support redirects for that case.

Reply:
I started with that, but changed it to this way. I think `opts` comes populated with some redundant things like the bucket name, possibly from some of the route-mangling middleware. So we'd then be in the business of stripping out the stuff we don't need, which would then just end up being some inverse of this. I thought it better to just test for these things.

Reply:
Ah fair enough, I hadn't realised there was other stuff in there.  Looks like it is not actually used though.